### PR TITLE
Sysroot merge ⛙ config

### PR DESCRIPTION
**Pull request by cgwalters**
Start of: https://github.com/ostreedev/ostree/issues/737

### CODE REVIEW

**Comment by cgwalters** [comment]
This will depend on https://github.com/ostreedev/ostree/pull/742

**Comment by cgwalters** [comment]
Rebased :surfing_man: and dropped the public API aspect - might still do that, but let's at least land the cleanups.

**Comment by cgwalters** [comment]
For after https://github.com/ostreedev/ostree/pull/865

File: src/libostree/ostree-sysroot-deploy.c [hunk 1]
```diff
@@ -807,26 +807,20 @@ merge_configuration (OstreeSysroot         *sysroot,
...
+  if (!ot_query_exists_at (deployment_dfd, "usr/etc", &usretc_exists, error))
+    return FALSE;
 
   if (etc_exists && usretc_exists)
-    {
-      g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
-                           "Tree contains both /etc and /usr/etc");
-      goto out;
-    }
+    return glnx_throw (error, "Tree contains both /etc and /usr/etc");
```
Comment (jlebon) [inline src/libostree/ostree-sysroot-deploy.c]: Hmm, this returned `NOT_FOUND` before, though `FAILED` seems more appropriate here anyway.

File: src/libostree/ostree-sysroot-deploy.c [hunk 2]
```diff
@@ -466,42 +484,37 @@ merge_etc_changes (GFile          *orig_etc,
   if (!ostree_diff_dirs (OSTREE_DIFF_FLAGS_IGNORE_XATTRS,
                          orig_etc, modified_etc, modified, removed, added,
                          cancellable, error))
-    {
-      g_prefix_error (error, "While computing configuration diff: ");
-      goto out;
-    }
+    return g_prefix_error (error, "While computing configuration diff: "), FALSE;
```
Comment (jlebon) [inline src/libostree/ostree-sysroot-deploy.c]: Nit: `glnx_prefix_error()`.

Comment (cgwalters) [inline src/libostree/ostree-sysroot-deploy.c]: OK, fixup :arrow_double_down:

**Comment by jlebon** [comment]
@rh-atomic-bot r+ 318e6dc

**Pull request closed without merging**